src/main/groovy/com/ullink/NuGetPack.groovy

```diff
@@ -86,14 +79,28 @@ class NuGetPack extends BaseNuGet {
         if (csprojPath) {
             return project.file(csprojPath)
         }
-        getNuSpecFile()
+        getNuspecFile()
     }
 
-    File getNuSpecFile() {
-        if (!this.nuspecFile || !project.file(this.nuspecFile).exists()) {
-            this.nuspecFile = generateNuspecFile()
+    GPathResult getNuspec() {
+        def nuspecFile = getNuspecFile()
+        if (nuspecFile) {
+            return new XmlSlurper().parse(project.file(nuspecFile))
+        }
+        def nugetSpec = getDependentNugetSpec()
+        if (nugetSpec) {
+            return new XmlSlurper(false, false).parseText(nugetSpec.generateNuspec())
+        }
+    }
+
+    File getNuspecFile() {
+        if (nuspecFile) {
+            return project.file(this.nuspecFile)
+        }
+        def nugetSpec = getDependentNugetSpec()
+        if (nugetSpec && nugetSpec.nuspecFile.exists()) {
```

Comment:
why the exists check ? it'll only work if the method is called after the task is executed, and I don't think it's necessary